catalog/transformer/catalog-transformer-zip/src/main/java/org/codice/ddf/catalog/transformer/zip/ZipCompression.java

```diff
@@ -13,261 +13,174 @@
+  private static final String METACARD_PATH = "metacards" + File.separator;
+
+  private static final String TRANSFORMER_ID = "transformerId";
+
+  private static final int BUFFER_SIZE = 4096;
 
-  private JarSigner jarSigner;
+  private List<ServiceReference> metacardTransformers;
 
-  public ZipCompression(JarSigner jarSigner) {
-    this.jarSigner = jarSigner;
+  private BundleContext bundleContext;
+
+  private static MimeType mimeType;
+
+  static {
+    try {
+      mimeType = new MimeType("application/zip");
+    } catch (MimeTypeParseException e) {
+      LOGGER.warn("Failed to apply mimetype application/zip");
+    }
+  }
+
+  public ZipCompression(BundleContext bundleContext, List<ServiceReference> metacardTransformers) {
+    this.bundleContext = bundleContext;
+    this.metacardTransformers = metacardTransformers;
   }
 
   /**
    * Transforms a SourceResponse with a list of {@link Metacard}s into a {@link BinaryContent} item
    * with an {@link InputStream}. This transformation expects a key-value pair
    * "fileName"-zipFileName to be present.
    *
-   * @param upstreamResponse - a SourceResponse with a list of {@link Metacard}s to compress
+   * @param sourceResponse - a SourceResponse with a list of {@link Metacard}s to compress
    * @param arguments - a map of arguments to use for processing. This method expects "fileName" to
    *     be set
    * @return - a {@link BinaryContent} item with the {@link InputStream} for the Zip file
    * @throws CatalogTransformerException when the transformation fails
    */
   @Override
-  public BinaryContent transform(
-      SourceResponse upstreamResponse, Map<String, Serializable> arguments)
+  public BinaryContent transform(SourceResponse sourceResponse, Map<String, Serializable> arguments)
       throws CatalogTransformerException {
 
-    checkArguments(upstreamResponse, arguments);
+    if (sourceResponse == null || CollectionUtils.isEmpty(sourceResponse.getResults())) {
+      throw new CatalogTransformerException(
+          "The source response does not contain any metacards to transform.");
+    }
 
-    String filePath = (String) arguments.get(ZipDecompression.FILE_PATH);
-    createZip(upstreamResponse, filePath);
+    if (arguments.get(TRANSFORMER_ID) == null) {
+      throw new CatalogTransformerException("Transformer ID cannot be null");
+    }
 
-    return getBinaryContentFromZip(filePath);
-  }
+    String transformerId = arguments.getOrDefault(TRANSFORMER_ID, "").toString();
 
-  private void checkArguments(SourceResponse upstreamResponse, Map<String, Serializable> arguments)
-      throws CatalogTransformerException {
-    if (upstreamResponse == null || CollectionUtils.isEmpty(upstreamResponse.getResults())) {
-      throw new CatalogTransformerException("No Metacards were found to transform.");
+    if (StringUtils.isBlank(transformerId)) {
+      throw new CatalogTransformerException("A valid transformer ID must be provided.");
```

Comment:
❓ If the old behavior did not throw an exception when this was blank, wont this change in behavior potentially break 'legacy' calls to this? Should we default blank to mean the old behavior of using object serialization instead of throwing an exception?